jetty-websocket/websocket-jetty-common/src/main/java/org/eclipse/jetty/websocket/common/JettyWebSocketRemoteEndpoint.java

```diff
@@ -265,10 +254,14 @@ public SocketAddress getRemoteAddress()
     @Override
     public void flush() throws IOException
     {
-        try (SharedBlockingCallback.Blocker b = blocker.acquire())
-        {
-            coreSession.flush(b);
-            b.block();
-        }
+        FutureCallback b = new FutureCallback();
+        coreSession.flush(b);
+        b.block(getBlockingTimeout(), TimeUnit.MILLISECONDS);
+    }
+
+    private long getBlockingTimeout()
```

Comment:
I'm wondering if we should repeat the pattern (mistake?) of HTTP here.  With HTTP we have a separately configurable blocking timeout, that only if not set do we default to idleTimeout.     This way we can handle (if need be) the timeouts for sending a huge frame that might take longer than the idle timeout, even though it is never actually idle.

However, in HTTP, we have deprecated the blocking timeout in favour of the data rate mechanisms....  Which we don't really have here (but are probably good ideas for @ extensions?), but it may still be good to have to turn off the blocking timeout if we have confidence in our callback mechanism.

@sbordet - thoughts?  I'm only making this a comment as I don't think we should hold this PR up while we ponder this.